delphi-checks/src/main/java/au/com/integradev/delphi/checks/RedundantJumpCheck.java

```diff
@@ -69,19 +63,20 @@ private void checkBlock(Block block, DelphiCheckContext context) {
     }
 
     UnconditionalJump jump = (UnconditionalJump) block;
-    Block successorWithoutJump = jump.getSuccessorIfRemoved();
     DelphiNode terminator = jump.getTerminator();
 
     RoutineNameDeclaration routineNameDeclaration = getRoutineNameDeclaration(terminator);
+    if (routineNameDeclaration == null) {
+      return;
+    }
+
     if (!isContinueOrExit(routineNameDeclaration)
         || isExitWithExpression(routineNameDeclaration, terminator)) {
       return;
     }
 
     Block successor = jump.getSuccessor();
-    successorWithoutJump = nonEmptySuccessor(successorWithoutJump);
```

Comment:
I'm a little nervous about this assumption that `UnconditionalJump` can never be followed by a `Linear` block. `UnconditionalJump` occurs many times in a CFG, most notably for every loop statement. Are we certain that this case can never come up?

Reply:
@jgardn3r and I looked over the CFG and concluded that the `nonEmptySuccessor` we're looking for here is basically mythical. See `ControlFlowGraphImpl::prune`, where empty linear blocks are stripped from the graph.

Reply:
Sure, if it's cosigned by Josh 🫡